ui/components/ui/identicon/identicon.component.js

```diff
@@ -78,16 +88,25 @@ export default class Identicon extends PureComponent {
   }
 
   render() {
-    const { address, image, useBlockie, addBorder, diameter } = this.props;
-
+    const {
+      address,
+      image,
+      useBlockie,
+      addBorder,
+      diameter,
+      useTokenDetection,
+      tokenList,
+    } = this.props;
     if (image) {
       return this.renderImage();
     }
 
     if (address) {
-      const checksummedAddress = toChecksumHexAddress(address);
-
-      if (checksummedAddress && contractMap[checksummedAddress]?.logo) {
+      // token from dynamic api list is fetched when useTokenDetection is true
+      const tokenAddress = useTokenDetection
+        ? address
+        : toChecksumHexAddress(address);
+      if (tokenAddress && tokenList[tokenAddress]?.iconUrl) {
```

Comment:
Could do a similar thing here: 
```
const tokenAddressKey = Object.keys(tokenList).find((tokenAddress) => isEqualCaseInsensitive(address, tokenAddress))

if (tokenAddress && tokenList[tokenAddressKey]?.iconUrl) {
...
```

Reply:
Won't be better to do a single check and conversion if needed, rather than looping through the whole list of tokens and converting them to lowercase every time the component is used??

Reply:
yeah you're right! 🙏